### PR TITLE
Created LoginStore, RegisterStore, AuthStore and UsersStore

### PR DESCRIPTION
**Pull request by andreytrofimov**

### CODE REVIEW

File: app/users-page/store/users.store.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    edit = (u: User) => {
+        this.gridState.edit(u);
+    };
+
+    @action
+    saveEdit = async (u: User) => {
+        this.serverError = '';
+
+        const r = await this.userApi.edit(u);
```
Comment (xarety) [inline app/users-page/store/users.store.ts:50]: beforeSave callback in the saveEdit could be used.

File: app/auth-pages/stores/register.store.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    strongPasswordValidator: Validator<string> = v =>
+        !FormValidators.passwordIsValidFormat(v)
+        && 'Your password must be at least 8 characters long including a number, ' +
+        'a lowercase letter, and an uppercase letter';
+
+    loginIsNotInUseValidator: Validator<string> = async v =>
+        (await this.isLoginAlreadyInUse(v)) && 'Login is already taken';
+
+    passwordMatchValidator: Validator<string> = v =>
```
Comment (xarety) [inline app/auth-pages/stores/register.store.ts]: password and confirm password could be grouped into one FormState to provide a generic validator.

File: app/auth-pages/stores/register.store.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        (this.form.$.passwordConfirmation.$ !== v) && 'Passwords must match';
+
+    passwordMatchConfirmationValidator: Validator<string> = v =>
+        (this.form.$.password.$ !== v) && ' ';
+
+    form = new FormState({
```
Comment (xarety) [inline app/auth-pages/stores/register.store.ts:45]: > **On form submission** must be checked next validation rules

File: app/users-page/store/users.store.ts (hunk 1 above)
Comment (andreytrofimov) [reply app/users-page/store/users.store.ts:50]: Looks like this will affect readability.

Comment (xarety) [reply app/users-page/store/users.store.ts:50]: Ok, sounds reasonable, but be careful, it won't work with the new editable grid.

File: app/auth-pages/stores/register.store.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    register = async () => {
+        this.serverError = '';
+        const res = await this.form.validate();
+        if (res.hasError) {
+            return;
+        }
+
+        const data = formStateToJS(this.form);
+
+        const r = await this.authApi.register(data);
```
Comment (xarety) [inline app/auth-pages/stores/register.store.ts]: Isn't now it have a different shape?

**Review by xarety — APPROVED** [review]

**Pull request merged**